src/app/metadata/services/datasetSummary.service.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * Short module description
+ * @copyright The GHGA Authors
+ * @license Apache-2.0
+ */
+
+import { HttpClient } from '@angular/common/http';
+import { computed, inject, Injectable, resource, signal, Signal } from '@angular/core';
+import { ConfigService } from '@app/shared/services/config.service';
+import { firstValueFrom } from 'rxjs';
+import { DatasetSummary, emptyDatasetSummary } from '../models/dataset-summary';
+
+/**
+ * Dataset summary service
+ *
+ * This service provides the functionality to fetch dataset summary data from the server.
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class DatasetSummaryService {
+  #http = inject(HttpClient);
+
+  #config = inject(ConfigService);
+  #metlDataUrl = this.#config.metldataURL;
+  #id_ = signal<string | undefined>(undefined);
+
+  #datasetSummary = resource({
+    request: computed(() => ({
+      id_: this.#id_(),
+    })),
+    loader: (param) => {
+      const id_ = param.request.id_;
+      if (id_) {
+        return Promise.resolve(
+          firstValueFrom(
+            this.#http.get<DatasetSummary>(
+              `${this.#metlDataUrl}/artifacts/stats_public/classes/DatasetStats/resources/${id_}`,
```

Comment:
You can prepare the URL beforehand:

```js
#datasetSummaryURL = `${this.#metlDataUrl}/artifacts/stats_public/classes/DatasetStats/resources`
```

and then use it here:
```js
this.#http.get<DatasetSummary>(`${this.#datasetSummaryUrl/${id_}`)
```

Reply:
When the services are merged together, the URL definitions should then appear at one place.

Reply:
Done 👍

Reply:
Not yet done in metadata.service

Reply:
It is indeed.

```
    loader: (param) => {
      const id_ = param.request.id_;

      const datasetSummaryURL = `${this.#metldataURL}/artifacts/stats_public/classes/DatasetStats/resources/${id_}`;```

Reply:
Ok, but the idea was to initialize that URL upfront as a class level property, directly after the line that defines `#globalSummaryUrl`.

Reply:
But wouldn't that mess with the parameter values not being dynamic?

Reply:
Just the first part of the URL